src/main/scala/cz/alenkacz/db/postgresscala/PostgresAsyncDbValue.scala

```diff
@@ -1,6 +1,81 @@
 package cz.alenkacz.db.postgresscala
+import java.net.InetAddress
+import java.sql.Time
+import java.time.Instant
+import java.util.UUID
+
+import org.joda.time.{DateTime, DateTimeZone, LocalDateTime}
 
 class PostgresAsyncDbValue(value: Any) extends DbValue {
-  override def string(): String = value.asInstanceOf[String]
-  override def int(): Int = value.asInstanceOf[Int]
+  override def string: String = value.asInstanceOf[String]
+
+  override def int: Int = value.asInstanceOf[Int]
+
+  override def bigInt: BigInt = value.asInstanceOf[BigInt]
+
+  override def double: Double = value.asInstanceOf[Double]
+
+  override def float: Float = value.asInstanceOf[Float]
+
+  override def long: Long = value.asInstanceOf[Long]
+
+  override def bool: Boolean = value.asInstanceOf[Boolean]
+
+  override def short: Short = value.asInstanceOf[Short]
+
+  override def inetAddress: InetAddress = value.asInstanceOf[InetAddress]
+
+  override def uuid: UUID = value.asInstanceOf[UUID]
+
+  override def dateTime: Instant = value match {
+    case d: DateTime => Instant.ofEpochMilli(d.getMillis)
+    case d: LocalDateTime => Instant.ofEpochMilli(value.asInstanceOf[LocalDateTime].toDateTime(DateTimeZone.UTC).getMillis)
+    case _ => throw new IllegalArgumentException("Unsupported datetime type - expecting DateTime or LocalDatetime from joda library")
+  }
+
+  override def time: Time = value.asInstanceOf[Time]
+
+  override def bytes: Array[Byte] = value.asInstanceOf[Array[Byte]]
+
+  override def any: Any = value
+
+  override def stringOpt: Option[String] = Option(string)
+
+  override def strings: Array[String] = value.asInstanceOf[Array[String]]
```

Comment:
Are you sure that retype to `Array[String]` works? Just asking, maybe a test would be nice.